src/main/java/com/zharnasek/alesya/game/PeopleDemo.java

```diff
@@ -14,22 +14,45 @@ public static void main(String[] args) {
 
         while (true) {
             System.out.print("Input name, pose or activity('exit' to stop): ");
-            String num = scanner.next();
+            String num = scanner.next().trim();
 
-            if (num.trim().equals("exit")) {
+            if (num.equals("exit")) {
                 break;
             }
 
-            if (num.trim().equals("pose")){
+            if (num.equals("pose")){
                 Random rand = new Random();
                 int randomIndex = rand.nextInt(gameResources.getPoses().size());
                 gameResources.getPoses().get(randomIndex).descrOutput();
             }
 
-            if (num.trim().equals("activity")){
-                Random rand = new Random();
-                int randomIndex = rand.nextInt(gameResources.getActivities().size());
-                gameResources.getActivities().get(randomIndex).descrOutput();
+            if (num.equals("activity")){
+                System.out.print("Input name for him: ");
+                Gamer gamerMale = new Gamer(scanner.next(), Gamer.Gender.MALE);
+                System.out.print("Input name for her: ");
+                Gamer gamerFemale = new Gamer(scanner.next(), Gamer.Gender.FEMALE);
+                String activitiesMale;
+                String activitiesFemale;
+                int randomIndex;
+
+                do {
+                    Random rand = new Random();
+                    do {
+                        randomIndex = rand.nextInt(gameResources.getActivities().size());
+                        activitiesFemale = gameResources.getActivities().get(randomIndex).toString();
+                    } while (!gameResources.getActivities().get(randomIndex).getGender().equals("FEMALE"));
+                    System.out.println(gamerFemale.getName() + ", " + activitiesFemale);
+                    System.out.print("input 'next' to continue, input 'exit' to finish: ");
+                    if (scanner.next().equals("next")) {
+                        do {
+                            randomIndex = rand.nextInt(gameResources.getActivities().size());
+                            activitiesMale = gameResources.getActivities().get(randomIndex).toString();
+                        } while (!gameResources.getActivities().get(randomIndex).getGender().equals("MALE"));
+                        System.out.println(gamerMale.getName() + ", " + activitiesMale);
+                    }
+                    System.out.print("input 'next' to continue, input 'exit' to finish: ");
+                } while (!scanner.next().equals("exit"));
+                flag = true;
             }
```

Comment:
I'm crying, not readable, and very complex

Reply:
updated